### PR TITLE
VB-3956 Exclude application reference from session window

### PR DESCRIPTION
**Pull request by AledWEvans**
I have pushed this up into the feature by mistake.

Please do not merge

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitSessionController.kt [hunk 1]
```diff
@@ -142,8 +142,14 @@ class VisitSessionController(
       example = "2024-12-20",
     )
     toDate: LocalDate,
+    @RequestParam(value = "excludeCurrentApplicationReference", required = false)
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitSessionController.kt:145]: name suggests its a boolean, could it be something like "excludedApplicationReference" ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt [hunk 2]
```diff
@@ -3,5 +3,5 @@ package uk.gov.justice.digital.hmpps.visitscheduler.dto.enums
 @Suppress("unused")
 enum class SessionConflict {
   NON_ASSOCIATION,
-  DOUBLE_BOOKED,
+  DOUBLE_RESERVATION,
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt:6]: Question: will this require a frontend change, do they implement this enum?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt:6]: will need updating on the Orchestration Service.

**Review by dhirajshettymoj — APPROVED** [review]
nothing other than @adaviesMOJ's comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt (hunk 2 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt:6]: DOUBLE_BOOKING_OR_RESERVATION

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt [hunk 3]
```diff
@@ -3,5 +3,5 @@ package uk.gov.justice.digital.hmpps.visitscheduler.dto.enums
 @Suppress("unused")
 enum class SessionConflict {
   NON_ASSOCIATION,
-  DOUBLE_BOOKED,
+  DOUBLE_BOOKING_OR_RESERVATION,
```
Comment (tpmcgowan) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/enums/SessionConflict.kt:6]: If we're renaming this (and I do think the existing name is misleading), then does `EXISTING_BOOKING_OR_RESERVATION` better describe it? For me 'double' implies that we've **already** got two there - but isn't it actually that there is **one** there and it's warning you of the risk of a second?

**Pull request merged**